### PR TITLE
[ADF-2131] Search sorting

### PR DESCRIPTION
**Pull request by DenisVuyka**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- search sorting
- update search results to better reflect mockups
- docs updates

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: demo-shell/src/app/components/files/files.component.ts [hunk 1]
```diff
@@ -74,6 +74,21 @@ export class FilesComponent implements OnInit, OnChanges, OnDestroy {
 
     processId;
 
+    @Input()
```
Comment (eromano) [inline demo-shell/src/app/components/files/files.component.ts]: All those input should have the jsdoc

File: demo-shell/src/app/components/search/search-result.component.ts [hunk 2]
```diff
@@ -79,4 +88,14 @@ export class SearchResultComponent implements OnInit {
     onDeleteElementSuccess(element: any) {
         this.searchResult.reload();
     }
+
+    private getSorting() {
```
Comment (eromano) [inline demo-shell/src/app/components/search/search-result.component.ts]: missing return type

File: lib/content-services/search/components/search-sorting-picker/search-sorting-picker.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SearchSortingPickerComponent } from './search-sorting-picker.component';
+import { SearchQueryBuilderService } from '../../search-query-builder.service';
+import { AppConfigService } from '@alfresco/adf-core';
+
+describe('SearchSortingPickerComponent', () => {
```
Comment (eromano) [inline lib/content-services/search/components/search-sorting-picker/search-sorting-picker.component.spec.ts]: Those tests are actually test the methods can we test the behaviours ? And do we need that this component is so bounded with the search instead to have a one general for the datatable?

File: demo-shell/src/app/components/files/files.component.ts (hunk 1 above)
Comment (DenisVuyka) [reply demo-shell/src/app/components/files/files.component.ts]: this one is demo shell and other properties don't have jsdoc there

Comment (eromano) [reply demo-shell/src/app/components/files/files.component.ts]: right 👍

**Comment by DenisVuyka** [comment]
@eromano fixed the flickering by providing the ability to switch off client-side sorting and trust whatever server gives back. Search controls the execution, provides the sorting settings and binds the result to a document list, so this works fine. For the whole navigation we may need a separate ticket and extending Document List with intelligent navigation when server-side sorting is used.

**Comment by DenisVuyka** [comment]
@eromano added a line for the new property. we can provide more details for 2.5 if we finish the whole server sorting story.

**Pull request merged**